packages/flutter/lib/src/painting/image_provider.dart

```diff
@@ -432,17 +432,42 @@ class NetworkImage extends ImageProvider<NetworkImage> {
     );
   }
 
-  static final http.Client _httpClient = createHttpClient();
+  static final HttpClient _httpClient = new HttpClient();
 
   Future<ui.Codec> _loadAsync(NetworkImage key) async {
     assert(key == this);
 
     final Uri resolved = Uri.base.resolve(key.url);
-    final http.Response response = await _httpClient.get(resolved, headers: headers);
-    if (response == null || response.statusCode != 200)
+    final HttpClientRequest request = await _httpClient.getUrl(resolved);
+    headers?.forEach((String name, String value) {
+      request.headers.add(name, value);
+    });
+    final HttpClientResponse response = await request.close();
+    if (response.statusCode != HttpStatus.OK)
       throw new Exception('HTTP request failed, statusCode: ${response?.statusCode}, $resolved');
 
-    final Uint8List bytes = response.bodyBytes;
+    final Completer<Uint8List> completer = new Completer<Uint8List>.sync();
+    if (response.contentLength == -1) {
+      final ByteConversionSink sink = new ByteConversionSink.withCallback((List<int> chunk) {
+        completer.complete(new Uint8List.fromList(chunk));
+      });
+      response.listen(sink.add, onDone: sink.close, onError: completer.completeError, cancelOnError: true);
+    } else {
+      final Uint8List bytes = new Uint8List(response.contentLength);
+      int offset = 0;
+      response.listen((List<int> chunk) {
+          for (int i = 0; i < chunk.length; i++, offset++) {
```

Comment:
nit: `+= 1` rather than `++`

Reply:
but forget that. Instead of this loop, just use `bytes.addAll(chunk)`.

Reply:
or rather, bytes.setRange

Reply:
Done